algorithms/CPlusPlus/Strings/ReverseTheStringWordwise.cpp

```diff
@@ -0,0 +1,44 @@
+// Description :- Given a string, the task is to reverse the order of the words in the given string. 
+// Example :-
+// Input 1:
+//    A = "the sky is blue"
+// Input 2:
+//    A = "this is ib"
+// Output 1:
+//    "blue is sky the"
+// Output 2:
+//    "ib is this"
+
+
+// Time Complexity = O(N), Space Complexity = O(N)
+string Solution::solve(string s) {
+   vector<string>v;
+        string str="";
+        for(int i=0;i<s.length();i++){
+            if(s[i]!=' '){
+                str+=s[i];
+            }
+            else if(str!="" && s[i]==' '){
+                v.push_back(str);
+                str="";
+            }
+        }
+        if(str!=""){
+            v.push_back(str);
+        }
+        str="";
+        for(int i=v.size()-1;i>0;i--){
+                str+=v[i];
+                str+=' ';
+        }
+        str+=v[0];
+        return str;
+}
+
+int main()
+{
+    string s;
+    cin>>s;
```

Comment:
`std::cin` stops reading once a space is encountered, the program is meant to be taking words for proper demonstration/testing, consider using `std::getline ()` instead
```suggestion
    std::getline (std::cin, s);
```